src/calliope/model.py

```diff
@@ -269,19 +264,14 @@ def build(
         else:
             backend_input = self._model_data
 
-        init_math_list = [] if self.config.build.ignore_mode_math else [mode]
-        end_math_list = [] if add_math_dict is None else [add_math_dict]
-        full_math_list = init_math_list + self.config.build.add_math + end_math_list
-        LOGGER.debug(f"Math preprocessing | Loading math: {full_math_list}")
-        model_math = preprocess.CalliopeMath(full_math_list, self.def_path)
-
+        self.applied_math = preprocess.build_applied_math(
```

Comment:
I would only attach this to `self` if the backend build succeeds as it hasn't actually been applied until then. Say that, attaching it first does act as a way to debug the resolved math definition if build fails...